### PR TITLE
Add 329 release notes

### PR DESCRIPTION
**Pull request by martint**
Closes https://github.com/prestosql/presto/issues/2473

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+Release 329 (XXX 2020)
+=========================
+
+General Changes
+---------------
+
+* Fix incorrect result of ``last_day_of_month`` function for first day of month. (:issue:`2452`)
+* Handle common disk failures during the spill. When on of disks got broken, it Presto will use another
+  one without failing the query. (:issue:`2444`)
+* Reduce number of metadata queries when querying ``information_schema``. (:issue:`2488`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]: end-user visible impact is "faster" or "improve performance"; maybe also "reduce metastore load":
```
* Improve performance and reduce load on external systems when querying ``information_schema``. (:issue:`2488`)
```

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+* Fix incorrect data returned when using S3 Select on uncompressed files. In our testing, S3 Select
+  was apparently returning incorrect results when reading uncompressed files, so S3 Select is disabled
+  for uncompressed files. (:issue:`2399`)
+* Fix incorrect data returned when using S3 Select on a table with ``skip.header.line.count`` or
+  ``skip.footer.line.count`` property. S3 Select API does not support skipping footers or more than one
+  line of a header.  In our testing, S3 Select was apparently sometimes returning incorrect results when
+  reading a compressed file with header skipping, so S3 Select is disabled when any of these table
+  properties is set to non-zero value. (:issue:`2399`)
+* Fix ``INSERT`` or ``CREATE TABLE ... AS`` query failure when one of the inserted ``REAL`` or ``DOUBLE`` values
+  is infinite or ``NaN``. (:issue:`2471`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]:
```suggestion
  is infinite or ``NaN`` and column statistics collection is not disabled. (:issue:`2471`)
```

(it's on by default)

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+* Add explicit support for conversion from Parquet decimal column type to Presto integer column types. (:issue:`2451`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]:
```
* Support reading data when column is declared as an integer type on partition level but is a decimal type
  in Parquet file. (:issue:`2451`)
```

@sopel39 is this a regression fix?

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+* Add validation of Parquet decimal column type scale. (:issue:`2451`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]: @sopel39 what's end-user visible effect?

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+* Delete HDFS location when dropping an empty schema. (:issue:`2463`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]: @ebyhr is it specific to HDFS? does it apply to S3 or other storages too?

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+* Execute table and partition drops in parallel. (:issue:`1812`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]:
```
* Improve performance when deleting multiple partitions of a table or multiple partitions in a transaction
  by executing these actions concurrently. (:issue:`1812`)
```

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+* Use listobject v2 of S3 API. (:issue:`2236`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]: @sopel39 what's end-user visible effect?

**Comment by electrum** [comment]
I added one more note about ``information_schema.columns`

File: presto-docs/src/main/sphinx/release/release-329.rst (hunk 7 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-329.rst]: I don't think there is any end user visible effect.

File: presto-docs/src/main/sphinx/release/release-329.rst (hunk 5 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-329.rst]: It applies to all storage locations, so we can replace "HDFS" with "storage"

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+Security Changes
+----------------
+
+* Add ``ldap.bind-dn`` and ``ldap.bind-password`` LDAP properties to allow LDAP authentication
+  access LDAP server using service account. (:issue:`1917`)
+
+CLI Changes
+-----------
+
+* Allow inspecting network requests. (:issue:`2524`)
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-329.rst]: How?

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 9]
```diff
@@ -0,0 +1,58 @@
+=========================
+Release 329 (XXX 2020)
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-329.rst]: This needs to be updated before release

File: presto-docs/src/main/sphinx/release/release-329.rst (hunk 8 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-329.rst]: By enabling DEBUG on `io.prestosql.cli.QueryRunner` via CLI's `--log-levels-file` switch

(`presto-cli --log-levels-file <(echo "io.prestosql.cli.QueryRunner=DEBUG")`)

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+* Handle common disk failures during spill. When one of disks got broken, it
+  Presto will use another one without failing the query. (:issue:`2444`)
+* Improve performance and reduce load on external systems when
+  querying ``information_schema``. (:issue:`2488`)
+* Limit broadcasted table size to 100MB by default when using the ``AUTOMATIC``
+  join type selection strategy. (:issue:`2527`)
+* Hide non-standard columns ``comment`` and ``extra_info`` in the standard
+  ``information_schema.columns`` table. These columns can still be selected,
+  but will no longer appear when describing the table. (:issue:`2306`)
+* Enable ``AUTOMATIC`` join reordering and join type selection strategy by default. (:issue:`2528`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]: I think it would be more informative if supplemented with "CBO" and "Cost-Based Optimizer".
Otherwise this is very easy to overlook

Comment (martint) [reply presto-docs/src/main/sphinx/release/release-329.rst]: I don't like using the term CBO -- it's too opaque. And "cost-based optimizer" is a little too generic, since it's not a general purpose cost-based optimizer. It only applies to join reordering and type selection. Maybe we can just link to the corresponding sections under https://prestosql.io/docs/current/optimizer/cost-based-optimizations.html

File: presto-docs/src/main/sphinx/release/release-329.rst (hunk 8 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-329.rst]: maybe add "... by enabling DEBUG logging for `io.prestosql.cli.QueryRunner`" ?

Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-329.rst]: good idea

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 11]
```diff
@@ -0,0 +1,58 @@
+=========================
+Release 329 (XXX 2020)
+=========================
+
+General Changes
+---------------
+
+* Fix incorrect result of ``last_day_of_month`` function for first day of month. (:issue:`2452`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-329.rst]: Link to function and say "for" instead of "of"
```
* Fix incorrect result for :func:`last_day_of_month` function for first day of month. (:issue:`2452`)

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+* Handle common disk failures during spill. When one of disks got broken, it
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-329.rst]:
```
* Handle common disk failures during spill. When one disks fails, another
  disk will be used automatically without failing the query. (:issue:`2444`)

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+  Presto will use another one without failing the query. (:issue:`2444`)
+* Improve performance and reduce load on external systems when
+  querying ``information_schema``. (:issue:`2488`)
+* Limit broadcasted table size to 100MB by default when using the ``AUTOMATIC``
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-329.rst]: Code quotes around size value

File: presto-docs/src/main/sphinx/release/release-329.rst (hunk 8 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-329.rst]: This is not network requests, simply the URL. I think we should add first class support for printing network requests. I'm not sure this is even worth documenting.

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+
+* Fix incorrect data returned when using S3 Select on uncompressed files. In our testing, S3 Select
+  was apparently returning incorrect results when reading uncompressed files, so S3 Select is disabled
+  for uncompressed files. (:issue:`2399`)
+* Fix incorrect data returned when using S3 Select on a table with ``skip.header.line.count`` or
+  ``skip.footer.line.count`` property. S3 Select API does not support skipping footers or more than one
+  line of a header.  In our testing, S3 Select was apparently sometimes returning incorrect results when
+  reading a compressed file with header skipping, so S3 Select is disabled when any of these table
+  properties is set to non-zero value. (:issue:`2399`)
+* Fix ``INSERT`` or ``CREATE TABLE ... AS`` query failure when one of the inserted ``REAL`` or ``DOUBLE`` values
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-329.rst]:
```
* Fix query failure for writes when one of the inserted ``REAL`` or ``DOUBLE`` values
  is infinite or ``NaN`` and column statistics collection is enabled. (:issue:`2471`)

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+  is infinite or ``NaN`` and column statistics collection is not disabled. (:issue:`2471`)
+* Support reading data when column is declared as an integer type on partition level but is a decimal type
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-329.rst]:
```
* Allow reading data from Parquet files when the column type is declared as ``INTEGER``
  in the table or partition, but is a ``DECIMAL`` type in the file. (:issue:`2451`)
```

File: presto-docs/src/main/sphinx/release/release-329.rst (hunk 4 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-329.rst]: Based on the tests, I think this can be
```
* Validate the scale of decimal types when reading Parquet files. This prevents
  incorrect results when the decimal scale in the file does not match the declared
  type for the table or partition. (:issue:`2451`)

Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-329.rst]: Actually, this is part of the change above that allows reading these at all, so I'm not sure that it was possible to get incorrect results previously. Maybe it's correct to remove this?

File: presto-docs/src/main/sphinx/release/release-329.rst (hunk 3 above)
Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-329.rst]: @findepi It wasn't explicitly supported (by coincidence really), but yes, it worked before.

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+  in Parquet file. (:issue:`2451`)
+* Add validation of Parquet decimal column type scale. (:issue:`2451`)
+* Delete storage location when dropping an empty schema. (:issue:`2463`)
+* Improve performance when deleting multiple partitions of a table or multiple partitions in a transaction
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-329.rst]: Maybe shorten this to

> Improve performance when deleting multiple partitions by executing these actions concurrently.

File: presto-docs/src/main/sphinx/release/release-329.rst (hunk 4 above)
Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-329.rst]: Maybe something like this?
`Prevent incorrect reading of decimal values from Parquet decimal columns, which don't match partition level decimal column type`

Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-329.rst]: > Actually, this is part of the change above that allows reading these at all, so I'm not sure that it was possible to get incorrect results previously. Maybe it's correct to remove this?

It was possible as long as both decimals were of the same type (e.g short or long). Parquet code was just inserting unscaled values...

File: presto-docs/src/main/sphinx/release/release-329.rst (hunk 8 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-329.rst]: don't have strong opinion. I'm fine either way

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 17]
```diff
@@ -0,0 +1,61 @@
+=========================
+Release 329 (XXX 2020)
+=========================
+
+General Changes
+---------------
+
+* Fix incorrect result for :func:`last_day_of_month` function for first day of month. (:issue:`2452`)
+* Fix incorrect results when handling ``DOUBLE`` or ``REAL`` types with ``NaN`` values. (:issue:`2582`)
+* Handle common disk failures during spill. When one disks fails, another
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]: disks -> disk

File: presto-docs/src/main/sphinx/release/release-329.rst (hunk 8 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-329.rst]: Ok, I'll remove it then

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+Release 329 (XXX 2020)
+=========================
+
+General Changes
+---------------
+
+* Fix incorrect result for :func:`last_day_of_month` function for first day of month. (:issue:`2452`)
+* Fix incorrect results when handling ``DOUBLE`` or ``REAL`` types with ``NaN`` values. (:issue:`2582`)
+* Handle common disk failures during spill. When one disk fails, another
+  disk will be used automatically without failing the query. (:issue:`2444`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]: This is promising too much. Maybe

```
* Handle common disk failures during spill. When one disk fails but multiple
  spill locations are configured, the healthy disks will be used for future queries.
  (:issue:`2444`)
```

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+  querying ``information_schema``. (:issue:`2488`)
+* Limit broadcasted table size to ``100MB`` by default when using the ``AUTOMATIC``
+  join type selection strategy. (:issue:`2527`)
+* Hide non-standard columns ``comment`` and ``extra_info`` in the standard
+  ``information_schema.columns`` table. These columns can still be selected,
+  but will no longer appear when describing the table. (:issue:`2306`)
+* Enable :doc:`cost based </optimizer/cost-based-optimizations>` join reordering and join type selection
+  optimizations by default. The previous behavior can be restored by
+  setting ``optimizer.join-reordering-strategy`` configuration property to ``ELIMINATE_CROSS_JOINS``
+  and ``join-distribution-type`` to ``PARTITIONED``. (:issue:`2528`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]: Move this next to (just above?) "Limit broadcasted table size" bullet

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+---------------
+
+* Fix incorrect result for :func:`last_day_of_month` function for first day of month. (:issue:`2452`)
+* Fix incorrect results when handling ``DOUBLE`` or ``REAL`` types with ``NaN`` values. (:issue:`2582`)
+* Handle common disk failures during spill. When one disk fails, another
+  disk will be used automatically without failing the query. (:issue:`2444`)
+* Improve performance and reduce load on external systems when
+  querying ``information_schema``. (:issue:`2488`)
+* Limit broadcasted table size to ``100MB`` by default when using the ``AUTOMATIC``
+  join type selection strategy. (:issue:`2527`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]: end-user impact could be described as:
```
... This avoid query failures or excessive memory usage when joining two or more very large tables.
```

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+* Fix incorrect data returned when using S3 Select on uncompressed files. In our testing, S3 Select
+  was apparently returning incorrect results when reading uncompressed files, so S3 Select is disabled
+  for uncompressed files. (:issue:`2399`)
+* Fix incorrect data returned when using S3 Select on a table with ``skip.header.line.count`` or
+  ``skip.footer.line.count`` property. S3 Select API does not support skipping footers or more than one
+  line of a header.  In our testing, S3 Select was apparently sometimes returning incorrect results when
+  reading a compressed file with header skipping, so S3 Select is disabled when any of these table
+  properties is set to non-zero value. (:issue:`2399`)
+* Fix query failure for writes when one of the inserted ``REAL`` or ``DOUBLE`` values
+  is infinite or ``NaN`` and column statistics collection is enabled. (:issue:`2471`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]: Originally i worded this as "and column statistics collection is not disabled", because it's enabled by default.

Maybe we should just omit that part?

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+-------------------------------
+
+* Add support for :ref:`elasticsearch-array-types`. (:issue:`2441`)
+* Reduce load on Elasticsearch cluster and improve query performance. (:issue:`2561`)
+
+PostgreSQL Connector Changes
+----------------------------
+
+* Fix mapping between PostgreSQL's ``TIME`` and Presto's ``TIME`` data types.
+  Previously the mapping was converting values incorrectly, shifting it by relative offset between
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-329.rst]:
```suggestion
  Previously the mapping was incorrect, shifting the values by relative offset between
```

File: presto-docs/src/main/sphinx/release/release-329.rst (hunk 21 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-329.rst]: The phrase "is not disabled" sounds like a double negative, but I see your point. Omit sounds good.

File: presto-docs/src/main/sphinx/release/release-329.rst [hunk 23]
```diff
@@ -56,3 +56,10 @@ Elasticsearch Connector Changes
 
 * Add support for :ref:`elasticsearch-array-types`. (:issue:`2441`)
 * Reduce load on Elasticsearch cluster and improve query performance. (:issue:`2561`)
+
+PostgreSQL Connector Changes
+----------------------------
+
+* Fix mapping between PostgreSQL's ``TIME`` and Presto's ``TIME`` data types.
+  Previously the mapping was converting values incorrectly, shifting it by relative offset between
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-329.rst]: > by the relative offset between the session zone and the Presto server's JVM zone
